Build fails with @tiptap/pm v3 — Module not found '@tiptap/pm/collab'

I upgraded to `@tiptap/pm@3.22.4` (and `@tiptap/core@3.22.4`) and now my Next.js build fails with this error:                                                                    
   
  ```                                                                                                                                                                              
  ./node_modules/.pnpm/@convex-dev+prosemirror-sync@0.2.3_.../dist/tiptap/index.js:3:1
  Module not found: Can't resolve '@tiptap/pm/collab'                                                                                                                              
   
    1 | import { useConvex, useMutation, useQuery } from "convex/react";                                                                                                           
    2 | import { Extension } from "@tiptap/core";                                                                                                                                  
  > 3 | import * as collab from "@tiptap/pm/collab";                                                                                                                               
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                               
    4 | import { Step } from "@tiptap/pm/transform";                                                                                                                               
  ```                                                                                                                                                                              
                                                                                                                                                                                   
  The error trail goes: my doc editor → `@convex-dev/prosemirror-sync/blocknote` → `@convex-dev/prosemirror-sync/tiptap`.                                                          
 


## Why it breaks                                                                                                                                                                 
                                                            
  `@tiptap/pm` v3 **removed the `/collab` subpath export**. You can verify by checking its `package.json` exports — only these subpaths exist now:                                 
   
  ```                                                                                                                                                                              
  ./changeset, ./commands, ./dropcursor, ./gapcursor, ./history,
  ./keymap, ./model, ./state, ./tables, ./transform, ./view                                                                                                                        
  ```                                                                                                                                                                              
                                                                                                                                                                                   
  No `./collab`. In v2 it existed as a re-export of `prosemirror-collab`.                                                                                                          
                                                            
  The `peerDependencies` in this package say:                                                                                                                                      
  ```json                                                   
  "@tiptap/core": "^2.7.0 || ^3.10.7"                                                                                                                                              
  ```
                                                                                                                                                                                   
  So v3 of tiptap is supposedly supported, but `dist/tiptap/index.js` still imports from a path that only existed in v2. 

## My setup                                                                                                                                                                      
                                                            
  - `@convex-dev/prosemirror-sync@0.2.3`                                                                                                                                           
  - `@tiptap/pm@3.22.4`
  - `@tiptap/core@3.22.4`                                                                                                                                                          
  - BlockNote-based editor (`@blocknote/core@0.48.x`)       
  - Next.js 16.2.4 with Turbopack                                                                                                                                                  
  - pnpm monorepo (Turborepo)

## Suggested fix                                                                                                                                                                 
                                                            
  Replace the broken import in `src/tiptap/index.ts`:                                                                                                                              
   
  ```diff                                                                                                                                                                          
  - import * as collab from "@tiptap/pm/collab";            
  + import * as collab from "prosemirror-collab";
  ```                                                                                                                                                                              
   
  …and add `prosemirror-collab` as a direct dependency (or peer dep). That's what `@tiptap/pm/collab` was re-exporting under the hood in v2 anyway, so the behavior is identical.  
                                                            
  ## My workaround for now                                                                                                                                                         
                                                            
  Until a fix is shipped, I'm aliasing the bad import in `next.config.ts`:                                                                                                         
   
  ```ts                                                                                                                                                                            
  const nextConfig: NextConfig = {                          
    turbopack: {
      resolveAlias: {
        "@tiptap/pm/collab": "prosemirror-collab",
      },                                                                                                                                                                           
    },
  }                                                                                                                                                                                
  ```                                                       

  …with `prosemirror-collab` installed as a direct dep on the app. Build passes after that.

--- a/package.json
+++ b/package.json
@@ -69,6 +69,9 @@
       "default": "./dist/component/convex.config.js"
     }
   },
+  "dependencies": {
+    "prosemirror-collab": "^1.3.1"
+  },
   "peerDependencies": {
     "@blocknote/core": ">=0.31.0 <=1.0.0",
     "@blocknote/mantine": ">=0.31.0 <=1.0.0",

--- a/src/tiptap/index.ts
+++ b/src/tiptap/index.ts
@@ -12,7 +12,7 @@ import {
   Extension,
   type JSONContent,
 } from "@tiptap/core";
-import * as collab from "@tiptap/pm/collab";
+import * as collab from "prosemirror-collab";
 import { Step } from "@tiptap/pm/transform";
 import { useCallback, useMemo, useRef } from "react";
 import type { SyncApi } from "../client/index.js";

